src/main/java/org/hisp/dhis/tasks/AddEventsTask.java

```diff
@@ -0,0 +1,95 @@
+package org.hisp.dhis.tasks;
+
+import static com.google.api.client.http.HttpStatusCodes.STATUS_CODE_OK;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.hisp.dhis.actions.RestApiActions;
+import org.hisp.dhis.cache.EntitiesCache;
+import org.hisp.dhis.dxf2.events.event.Event;
+import org.hisp.dhis.random.EventRandomizer;
+import org.hisp.dhis.random.RandomizerContext;
+import org.hisp.dhis.response.dto.ApiResponse;
+import org.hisp.dhis.utils.DataRandomizer;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * @author Luciano Fiandesio <luciano@dhis2.org>
+ */
+public class AddEventsTask
+    extends DhisAbstractTask
+{
+    private EntitiesCache entitiesCache;
+    private EventRandomizer eventRandomizer;
+    private String endpoint = "/api/events";
+    public AddEventsTask(int weight, EntitiesCache entitiesCache )
+    {
+        this.weight = weight;
+        this.entitiesCache = entitiesCache;
+        eventRandomizer = new EventRandomizer();
+    }
+
+    @Override
+    public String getName()
+    {
+        return "POST events";
+    }
+
+    private Event createRandomEvent()
+    {
+        try
+        {
+            return eventRandomizer.create( entitiesCache, RandomizerContext.EMPTY_CONTEXT() );
+        }
+        catch ( Exception e )
```

Comment:
I think we shouldn't catch exception here and instead let it be handled by locust, show up in the dashboard.

Reply:
The possibility of an error here is related to a problem to the actual test, not to the performance results, so I don't think we should count it as a performance test error. I have added these checks mostly to avoid this problem that Locust would swallow the exception and stop reporting.

Reply:
Test errors have a separate tab in the locust dashboard. From locust4j: `Don't catch all exceptions in the execute method, just leave every unexpected exceptions to locust4j.`: https://github.com/myzhan/locust4j